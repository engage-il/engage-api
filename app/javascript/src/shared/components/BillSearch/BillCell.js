// @flow
import * as React from 'react'
import { createFragmentContainer, graphql } from 'react-relay'
import { css } from 'glamor'
import { formatDate } from 'shared/date'
import type { Bill } from 'shared/types'
import { Button } from 'shared/components'
import { stylesheet, colors, mixins } from 'shared/styles'

type Props = {
  bill: Bill,
  className?: string
}

let BillCell = function BillCell ({
  bill
}: Props) {
  const formattedDate = formatDate(bill.hearing.date, 'DD/MM/YYYY')

  return (
    <div {...css(rules.container)}>
      <div {...rules.info}>
        <div {...rules.header}>
          <div {...rules.document}>
            <h3>{bill.title}</h3>
            <span>{bill.documentNumber}</span>
          </div>
          <p>{formattedDate}</p>
        </div>
        {bill.summary && <p {...rules.summary}>{bill.summary}</p>}
      </div>
      <div {...rules.actions}>
        <Button
          styles={rules.button}
          to={bill.witnessSlipUrl}
          children='Take Action'
        />
        <Button
          styles={rules.button}
          to={`/bill/${bill.id}`}
          isSecondary
          children='More Info'
        />
      </div>
    </div>
  )
}

BillCell = createFragmentContainer(BillCell, graphql`
  fragment BillCell_bill on Bill {
    id
    documentNumber
    title
    summary
    witnessSlipUrl
    detailsUrl
    fullTextUrl
    hearing {
      date
    }
  }
`)

const rules = stylesheet({
  container: {
    ...mixins.shadows.low,
    ...mixins.borders.low(),
    display: 'flex',
    padding: 15,
    marginBottom: 15,
    backgroundColor: colors.neutral,
    '&:last-child': {
      marginBottom: 0
    },
    ...mixins.mobile.glam({
      flexDirection: 'column'
    })
  },
  info: {
    ...mixins.borders.low(['right']),
    flex: 1,
    paddingRight: 15,
    ...mixins.mobile.glam({
      ...mixins.borders.low(['bottom']),
      paddingRight: 0,
      paddingBottom: 15,
      marginBottom: 15,
      borderRight: 'none'
    })
  },
  header: {
    display: 'flex',
    flexDirection: 'column'
  },
  document: {
    marginBottom: 5,
    '> *': {
      display: 'inline-block'
    },
    '> h3': {
      ...mixins.fonts.bold,
      marginRight: 10
    }
  },
  actions: {
    display: 'flex',
    flexDirection: 'column',
    paddingLeft: 15,
    ...mixins.mobile.glam({
      flexDirection: 'row',
      paddingLeft: 0
    })
  },
  button: {
    width: 200,
    marginBottom: 10,
    ':last-child': {
      marginBottom: 0
    },
    ...mixins.mobile.glam({
      flex: 1,
      width: 'auto',
      marginBottom: 0,
      marginRight: 10,
      ':last-child': {
        marginRight: 0
      }
    })
  },
  summary: {
    ...mixins.borders.low(['top']),
    marginTop: 10,
    paddingTop: 10
  }
})

export { BillCell }
