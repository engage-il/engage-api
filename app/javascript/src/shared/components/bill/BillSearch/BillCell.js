// @flow
import * as React from 'react'
import styled from 'react-emotion'
import { createFragmentContainer, graphql } from 'react-relay'
import { BillTitle } from '../BillTitle'
import { BillStatus } from '../BillStatus'
import { Button } from '@/components'
import { mixins } from '@/styles'
import type { Bill } from '@/types'

type Props = {
  bill: Bill,
  className?: string
}

let BillCell = function BillCell ({
  bill
}: Props) {
  return (
    <div>
      <Cell>
        <BillTitle bill={bill} />
        <BillStatus bill={bill} />
        <p>{bill.summary}</p>
        <Button to={`/bill/${bill.id}`} isSmall children='More Info' />
      </Cell>
    </div>
  )
}

BillCell = createFragmentContainer(BillCell, graphql`
  fragment BillCell_bill on Bill {
    id
    summary
    hearing {
      date
    }
    ...BillTitle_bill
    ...BillStatus_bill
  }
`)

const Cell = styled.div`
  ${mixins.flexColumn};

  align-items: flex-start;
  max-width: 960px;

  > p {
    margin-bottom: 15px;
  }
`

export { BillCell }
