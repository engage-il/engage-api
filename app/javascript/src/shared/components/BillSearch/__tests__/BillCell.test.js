/* eslint-env jest */
import { assign } from 'lodash'
import React from 'react'
import { shallow } from 'enzyme'
import { BillCell } from '../BillCell'

// subject
let subject
let bill

function loadSubject () {
  subject = shallow(<BillCell bill={bill} />).dive()
}

// specs
beforeEach(() => {
  subject = null
})

describe('#render', () => {
  beforeEach(() => {
    bill = {
      id: '1234',
      documentNumber: 'HB1234',
      title: 'Foo',
      summary: 'A bill, fantastic',
      witnessSlipUrl: 'http://www.test.com/slip',
      detailsUrl: 'http://www.test.com/details',
      fullTextUrl: 'http://www.test.com/text',
      hearing: {
        date: '2010-01-01T00:00:00-06:00'
      }
    }
  })

  it('renders properly', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })

  it('hides the summary when missing', () => {
    assign(bill, { summary: '' })
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})

describe('the relay container', () => {
  it('exists', () => {
    expect(BillCell.container).toBeTruthy()
  })
})
