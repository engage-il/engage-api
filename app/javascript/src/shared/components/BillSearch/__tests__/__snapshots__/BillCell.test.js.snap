// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render hides the summary when missing 1`] = `
<Styled(div)>
  <Styled(div)>
    <Styled(img)
      alt="Bill Icon"
      src={null}
    />
    <p>
      HB1234
       - Updated 
      Invalid Date
    </p>
  </Styled(div)>
  <h3>
    Foo
  </h3>
  <Styled(div) />
  <p />
  <Button
    isSmall={true}
    to="/bill/1234"
  >
    More Info
  </Button>
</Styled(div)>
`;

exports[`#render renders properly 1`] = `
<Styled(div)>
  <Styled(div)>
    <Styled(img)
      alt="Bill Icon"
      src={null}
    />
    <p>
      HB1234
       - Updated 
      Invalid Date
    </p>
  </Styled(div)>
  <h3>
    Foo
  </h3>
  <Styled(div) />
  <p>
    A bill, fantastic
  </p>
  <Button
    isSmall={true}
    to="/bill/1234"
  >
    More Info
  </Button>
</Styled(div)>
`;
